core/src/main/java/org/apache/carbondata/core/util/CleanFilesUtil.java

```diff
@@ -163,8 +164,13 @@ private static void getStaleSegmentFiles(CarbonTable carbonTable, List<String> s
     }
     Set<String> loadNameSet = Arrays.stream(details).map(LoadMetadataDetails::getLoadName)
         .collect(Collectors.toSet());
-    List<String> staleSegments = segmentFiles.stream().filter(segmentFile -> !loadNameSet.contains(
-        DataFileUtil.getSegmentNoFromSegmentFile(segmentFile))).collect(Collectors.toList());
+    // get all stale segment files, not include compaction segments
```

Comment:
we don't want to send compacted segments to the trash folder?

Reply:
please add a detail comments about why we need to ignore stale compacted (x.y) segment.

Reply:
done